src/inmanta/deploy/scheduler.py

```diff
@@ -464,9 +552,10 @@ async def _new_version(
             added_requires: dict[ResourceIdStr, Set[ResourceIdStr]] = {}
             dropped_requires: dict[ResourceIdStr, Set[ResourceIdStr]] = {}
             for resource, details in up_to_date_resources.items():
-                self._state.add_up_to_date_resource(resource, details)
+                self._state.add_up_to_date_resource(resource, details)  # Removes from the dirty set
 
             for resource, details in resources.items():
+                await self.create_periodic_repair_and_deploy(resource)
```

Comment:
This is not correct right?

This timer is wrt to when the resource as first seen, not wrt to the last time it was deployed?

I would expect a non-periodic timer that reset on every deploy?

Reply:
Correction: I would start timer only for new resources, not for new versions.